cmd/stack_create.go

```diff
@@ -83,54 +87,93 @@ The stack name must start with a lowercase letter, and can contain only lowercas
 				}
 			}
 
-			err = downloadFileToDisk(rootConfig.LoggingConfig, "https://github.com/appsody/stacks/archive/master.zip", filepath.Join(getHome(rootConfig), "extract", "repo.zip"), config.Dryrun)
+			repoID, stackID, err := parseProjectParm(config.copy, config.RootCommandConfig)
 			if err != nil {
 				return err
 			}
-			_, stackTempDir, err := parseProjectParm(config.copy, config.RootCommandConfig)
+
+			switch repoID {
+			case "incubator":
+				repoID = "incubator-index"
+			case "experimental":
+				repoID = "experimental-index"
+			default:
+			}
+
+			extractFilename := stackID + ".tar.gz"
+			extractDir := filepath.Join(getHome(rootConfig), "extract")
+			extractDirFile := filepath.Join(extractDir, extractFilename)
+
+			// Get Repository directory and umarshal
+			repoDir := getRepoDir(rootConfig)
+			var repoFile RepositoryFile
+			source, err := ioutil.ReadFile(filepath.Join(repoDir, "repository.yaml"))
 			if err != nil {
-				return err
+				return errors.Errorf("Error trying to read: %v", err)
 			}
 
-			valid, unzipErr := unzip(rootConfig.LoggingConfig, filepath.Join(getHome(rootConfig), "extract", "repo.zip"), stack, config.copy, config.Dryrun)
-			if unzipErr != nil {
-				return unzipErr
+			err = yaml.Unmarshal(source, &repoFile)
+			if err != nil {
+				return errors.Errorf("Error parsing the repository.yaml file: %v", err)
 			}
 
-			if !valid {
-				return errors.Errorf("Invalid stack name: " + config.copy + ". Stack name must be in the format <repo>/<stack>")
+			// get specificed repo and umarshal
+			repoInfo := repoFile.GetRepo(repoID)
+
+			// error if repo not found in repository.yaml
+			if repoInfo == nil {
+				//return errors.Errorf("Repository: %s not found in repository.yaml file", repoID)
+				return oldCreateMethod(rootConfig.LoggingConfig, rootConfig, config, stack, currentTime)
```

Comment:
This worked fine on my local branch, but on travis the test failed unless I returned the `oldCreateMethod` not sure if this needs to be reworked.

Reply:
Fixed now